rts/idris_utf8.c

```diff
@@ -78,6 +78,36 @@ unsigned idris_utf8_index(char* s, int idx) {
    return top;
 }
 
+char* idris_utf8_advance(char* str, int i) {
```

Comment:
Yeah, I noted that in the last review. a non utf-8 string (say a truncated string) can skip past the end.  Having looked at it some more, I think it should be changed. There is nothing that guarantees that what comes in from idris_readStr is a valid utf-8 string.

I would just take strlen at the beginning and check that it isn't exceeded. It's not the most performant way, but we can save microoptimization for later.